e2e/tools/validator/src/validator/cli/__init__.py

```diff
@@ -2,153 +2,201 @@
 #
 # SPDX-License-Identifier: APACHE-2.0
 
-import click
 import subprocess
 import os
+import logging
+import typing
+import json
+
+import click
+from prometheus_api_client.utils import parse_datetime
+
 from validator.__about__ import __version__
-from validator.stresser import ( Remote )
+from validator.config import Validator, load
+from validator.specs import Reporter as SpecReporter
+from validator.stresser import Remote
+
+from validator.stresser import ScriptResult
+from validator.validations import Loader, QueryTemplate
+from validator.prometheus import Comparator, Series, PrometheusClient
 
-from validator.prometheus import MetricsValidator
+logger = logging.getLogger(__name__)
+pass_config = click.make_pass_decorator(Validator)
 
-from validator.cases import Cases
 
-from validator.config import (
-    Validator, load
-)
+class Report(typing.NamedTuple):
+    path: str
+    results_dir: str
+    file: typing.TextIO
 
-from validator.specs import (
-    get_host_cpu_spec, get_vm_cpu_spec, get_host_dram_size, get_vm_dram_size
-)
+    def write(self, text: str) -> None:
+        self.file.write(text)
+
+    def close(self) -> None:
+        self.file.close()
+
+    def flush(self) -> None:
+        self.file.flush()
+
+    def h1(self, text: str) -> None:
+        self.write(f"# {text}\n\n")
+
+    def h2(self, text: str) -> None:
+        self.write(f"## {text}\n\n")
+
+    def h3(self, text: str) -> None:
+        self.write(f"### {text}\n\n")
+
+    def h4(self, text: str) -> None:
+        self.write(f"#### {text}\n\n")
+
+
+def new_report() -> Report:
+    # run git describe command and get the output as the report name
+    git_describe = subprocess.run(["git", "describe", "--tag"], stdout=subprocess.PIPE)
+    tag = git_describe.stdout.decode().strip()
+
+    results_dir = os.path.join("/tmp", f"validator-{tag}")
+    os.makedirs(results_dir, exist_ok=True)
+
+    path = os.path.join("/tmp", f"report-{tag}.md")
```

Comment:
Can we use current directory for storing the results instead of `/tmp`?

Reply:
Lets add a reports param to `validator.yaml` in a different pr.